proposals/table-component.md

```diff
@@ -0,0 +1,221 @@
+    dataIndex: 1,
+  },
+  {
+    title: 'First',
+    dataIndex: 2,
+  },
+  {
+    title: 'Last',
+    dataIndex: 3,
+  }
+};
+
+export default () => <Table data={data} columns={columns}/>;
+```
+
+### Advanced Usage
+
+#### Sorting and custom rendering
+
+```javascript
+import React from 'react';
+import {Table, StyledCell} from 'baseui/table';
+
+const data = [
+  ['1', 'Sarah', 'Brown', 31, 'New York No. 1 Anywhere'],
+  ['2', 'Jane', 'Smith', 32, 'San Francisco No. 1 Anywhere'],
+  ['3', 'Joe', 'Black', 33, 'Sydney No. 1 Anywhere'],
+  ['4', 'Jane', 'Red', 34, 'London No. 1 Anywhere'],
+];
+
+const columns = [
+  {
+    title: 'Name',
+    sortComparator: (a, b) => {
+      return a[2] - b[2]; // sorts by last name
+    },
+    render: (rowData) => {
+      const [id, firstName, lastName] = rowData;
+      return (
+        <StyledCell>
+          `${firstName} ${lastName}`
+        </StyledCell>
+        );
+    }
+  },
+  {
+    title: 'Age',
+    dataIndex: 3,
+  },
+  {
+    title: 'Address',
+    dataIndex: 4,
+    sortComparator: (a, b) => a[4].length - b[4].length,
+  }
+];
+
+export default = () => <Table columns={columns} data={data}/>,
+```
+
+#### Remote data and pagination
+
+```javascript
+import * as React from 'react';
+import {Table} from 'baseui/table';
+import {Pagination} from 'baseui/pagination';
+
+const columns = {
+  {
+    title: 'Employee ID',
+    dataIndex: 1,
+  },
+  {
+    title: 'First',
+    dataIndex: 2,
+  },
+  {
+    title: 'Last',
+    dataIndex: 3,
+  }
+};
+
+class TableWithPagination extends React.Component {
+  state = {
+      currentPage: 1,
+      pageSize: 10,
+      data: [],
+      numPages: 0
+  };
+
+  fetch = (pageNumber, pageSize) => {
+      // logic to fetch data based on page number
+      // `currentPage` and `data` and `numPages` will be updated here.
+  }
+
+  componentDidMount = () => {
+      const {currentPage, pageSize} = this.state;
+      this.fetch(currentPage, pageSize);
+  }
+
+  onPageChange = (nextPage, prevPage) => {
+      const {pageSize} = this.state;
+      this.fetch(nextPage, pageSize);
+  }
+
+  render = () => {
+    return (
+        <Table
+          columns={columns}
+          dataSource={this.state.data}
+        />
+        <Pagination currentPage={this.state.currentPage} numPages={this.state.numPages} onPageChange={this.onPageChange}/>
+    );
+  }
+}
+```
+
+## Exports
+
+* `Table`
+* `StyledRoot`
+* `StyledHead`
+* `StyledHeadCell`
+* `StyledBody`
+* `StyledRow`
+* `StyledCell`
+
+## `Table` API
+
+* `columns: Array<{title: string, dataIndex: number, render?: (rowData: Array<any>) => React.Node, sortComparator?: (a: Array<any>, b: Array<any>) => number}>` - Required
+  * Describes how the column of data sorts and corresponding cells render.
+* `data: Array<Array<any>>` - Required
+  * Matrix data to be rendered within the table.
+* `estimatedRowSize?: number = 40px`
+  * Passed to `react-virtualized`. Used to estimate table height.
```

Comment:
@CharlieMartell @ericsoco interested in your input on the react-virtualized specific api

another is `useDynamicRowHeight` on L184